php/class-customize-snapshot-manager.php

```diff
@@ -334,6 +336,30 @@ public function enqueue_admin_scripts( $hook ) {
 	}
 
 	/**
+	 * Enqueue frontend scripts.
+	 *
+	 * These files control the behavior frontend.
+	 */
+	public function enqueue_frontend_scripts() {
+		if ( ! $this->snapshot ) {
```

Comment:
This also needs to short-circuit if the current user cannot publish changesets. In other words:

```php
if ( ! $this->snapshot || ! current_user_can( get_post_type_object( 'customize_changeset' )->cap->publish_posts ) ) {
```